reference/docs-conceptual/PSScriptAnalyzer/using-scriptanalyzer.md

```diff
@@ -0,0 +1,316 @@
+the rule functions with the verb `Measure`, so that is used for the property **IncludeRules**.
+
+```powershell
+@{
+    CustomRulePath      = @(
+        '.\output\RequiredModules\DscResource.AnalyzerRules'
+        '.\tests\QA\AnalyzerRules\SqlServerDsc.AnalyzerRules.psm1'
+    )
+
+    IncludeRules        = @(
+        'Measure-*'
+    )
+}
+```
+
+You can also add default rules by listing the rules in the **IncludeRules** property. When including
+default rules is important that the property **IncludeDefaultRules** is set to `$true`, otherwise
+the default rules are used.
+
+```powershell
+@{
+    CustomRulePath      = @(
+        '.\output\RequiredModules\DscResource.AnalyzerRules'
+        '.\tests\QA\AnalyzerRules\SqlServerDsc.AnalyzerRules.psm1'
+    )
+
+    IncludeDefaultRules = $true
+
+    IncludeRules        = @(
+        # Default rules
+        'PSAvoidDefaultValueForMandatoryParameter'
+        'PSAvoidDefaultValueSwitchParameter'
+
+        # Custom rules
+        'Measure-*'
+    )
+}
+```
+
+### Using custom rules in Visual Studio Code
+
+It is also possible to use the custom rules that are provided in the settings file in Visual Studio
+Code. This is done by adding a Visual Studio Code workspace settings file (`.vscode/settings.json`).
+
+```json
+{
+    "powershell.scriptAnalysis.settingsPath": ".vscode/analyzersettings.psd1",
+    "powershell.scriptAnalysis.enable": true,
+}
+```
+
+## ScriptAnalyzer as a .NET library
+
+ScriptAnalyzer engine and functionality can now be directly consumed as a library.
+
+Here are the public interfaces:
+
+```csharp
+using Microsoft.Windows.PowerShell.ScriptAnalyzer;
+
+public void Initialize(System.Management.Automation.Runspaces.Runspace runspace,
+Microsoft.Windows.PowerShell.ScriptAnalyzer.IOutputWriter outputWriter,
+[string[] customizedRulePath = null],
+[string[] includeRuleNames = null],
+[string[] excludeRuleNames = null],
+[string[] severity = null],
+[bool suppressedOnly = false],
+[string profile = null])
+
+public System.Collections.Generic.IEnumerable<DiagnosticRecord> AnalyzePath(string path,
+    [bool searchRecursively = false])
+
+public System.Collections.Generic.IEnumerable<IRule> GetRule(string[] moduleNames, string[] ruleNames)
+```
+
+## Violation Correction
+
+Some violations can be fixed by replacing the violation causing content with a suggested
+alternative. You can use the `-Fix` switch to automatically apply the suggestions. Since
+`Invoke-ScriptAnalyzer` implements **SupportsShouldProcess**, you can additionally use `-WhatIf` or
+`-Confirm` to find out which corrections would be applied. Be sure to use source control when
+applying those corrections since some changes, such as the one for **AvoidUsingPlainTextForPassword**,
+might require additional script modifications that cannot be made automatically. If your scripts are
+sensitive to encoding, you should also check for that because the initial encoding can not be
+preserved in all cases.
```

Comment:
```suggestion
You can use the **Fix** switch to to automatically replace violation-causing content with a
suggested alternative. Additionally, because `Invoke-ScriptAnalyzer` implements
**SupportsShouldProcess**, you can use **WhatIf** or **Confirm** to find out which corrections
would be applied. You should use source control when applying corrections as some changes, such as
the one for **AvoidUsingPlainTextForPassword**, might require additional script modifications that
can't be made automatically. Because initial encoding can't always be preserved when you
automatically apply suggestions, you should check your file's encoding if your scripts depend on a
particular encoding.
```